File: frontend/credit-wizard/src/app/shared/dtos/semesterPlannerModulDto.ts
import { Guid } from "guid-typescript";
import { ISemesterPlannnerDto } from "./semesterPlannerDto";
import { IModulDto } from "./modulDto";

export interface ISemesterPlannerModulDto {
    id: Guid,
    semesterPlannerDto: ISemesterPlannnerDto,
    modulId: Guid,
    modulDto: IModulDto,
    grade: number
}

File: backend/credit-wizard-api/credit-wizard-api/Dtos/SemesterPlannerModulDto.cs
using credit_wizard_api.Models;
using System.ComponentModel.DataAnnotations;

namespace credit_wizard_api.Dtos
{
    public class SemesterPlannerModulDto
    {
        [Required]
        public Guid SemesterPlannerId { get; set; }

        public SemesterPlannerDto? SemesterPlannerDto { get; set; }

        [Required]
        public Guid ModulId { get; set; }

        public ModulDto? ModulDto { get; set; }

        [Range(1, 6)]
        public double? Grade { get; set; }
    }
}